src/WebApiToTypeScript/Endpoints/AngularEndpointsService.cs

```diff
@@ -81,8 +109,10 @@ public void WriteServiceObjectToBlock(TypeScriptBlock serviceBlock, WebApiContro
                         )
                         .AddStatement($"var endpoint = new {endpointFullName}(args);")
                         .AddAndUseBlock("return _.extendOwn(endpoint,", isFunctionBlock: true, terminationString: ";")
-                        .AddAndUseBlock($"call<TView>({callArgumentDefinition})")
-                        .AddStatement($"return {Config.ServiceName}.call<TView>(this, {callArgumentValue});");
+                        .AddAndUseBlock($"call<TView>({callArgumentDefinition})", isFunctionBlock: false, terminationString: ",")
+                        .AddStatement($"return {Config.ServiceName}.call<TView>(this, {callArgumentValue});")
+                        .Parent
+                        .AddBlock(cachedBlock);
```

Comment:
Save this to a variable and do a `if (boolean) variable.AddAndUseBlock( //*create cached block here */)`?